compiler/daml-extension/src/webview.js

```diff
@@ -34,9 +44,28 @@ window.addEventListener('message', event => {
           document.body.classList.remove('hide_table');
           break;
         default:
-          console.log('Unexpected value for select_view: ' + message.value);
+          console.log('Unexpected value for select_view: ' + message.value.selected);
           break;
       }
+      if (message.value.showArchived) {
+        document.body.classList.remove('hide_archived');
+        document.body.classList.add('show_archived');
+        document.getElementById('show_archived').checked = true;
+      } else {
+        document.body.classList.remove('show_archived');
+        document.body.classList.add('hide_archived');
+        document.getElementById('show_archived').checked = false;
+      }
+      if (message.value.showDetailedDisclosure) {
+        document.body.classList.remove('hidden_disclosure');
+        document.body.classList.add('show_disclosure');
+        document.getElementById('show_detailed_disclosure').checked = true;
+      } else {
+        document.body.classList.remove('show_disclosure');
+        document.body.classList.add('hidden_disclosure');
+        document.getElementById('show_detailed_disclosure').checked = false;
+      }
+
```

Comment:
I think you can factor this out into a function that accepts 4 arguments:
1. The boolean that controls the state.
2. The class that is active when the boolean is true.
3. The class that is active when the boolean is false
4. The id of the checkbox.

Reply:
You could also try fitting the transaction vs table view in there. Not sure if that’s helpful or not.

Reply:
factored out in a function.